docs/dev/plugins/bash-script-generator/bash-script-generator.md

```diff
@@ -0,0 +1,16 @@
+# Bash script generator
+
+To provide the user a easy to use target artifact we provide him several build and deploy scripts. Since every plugin has to create those scripts is is better to build a generator that simplifies this job and ensures that the plugins do not produce duplicated code.
+
+## Idea 
+Provide a general so called **BashScriptGenerator** that can build bash scripts out of different components. A component could be the following snippet:
+```bash
+echo "Check if <executable> is available."                                                      
+if ! [ -x "$(command -v <executable>)" ]; then                                                 
+  echo 'Error: <executable> is not installed.' >&2                                             
+  exit 1                                                                                 
+fi     
+```
```

Comment:
Maybe it's better to provide functions in util scripts. When building the scripts, the plugins shall use these util functions, whenever possible. These util scripts need to be in the generated target artifact, of course.
This has the advantage of not generating a boilerplate of code (e.g., check for existence of 10 different executables -- would you rather like to have a) 10x above code or b) 10x `check <executable>` and 1x util script with function ?
A general generator would however still be needed for generating the header (`#!/bin/bash`) and for sourcing the util scripts.

Reply:
Forgott the bash functions, I am so stupid! So you want to give the user a general util script or a util script fitted to the targert platform?

Reply:
I'd ship a general util script

Reply:
I'd ship a general util script

Reply:
I suggest we use one general util script which is shipped with the target artifact.
The user does not have to interact with this script - however, the scripts called by the user might depend on this util script.

Platforms might create their own util scripts: e.g. `util-<platformId>.sh,`.

In order to have a nice java interface, every function should be reflected as java function (in the bash script generator).
This java function write's the call to the bash function, including necessary parameters, into the script.

Plugins could extend this ScriptGenerator and add provide custom java functions (with own util script).

Reply:
Had the same idea with the java interface. Thx for the input!